syncer/data_fetch.go

```diff
@@ -377,23 +383,35 @@ func (d *DataFetch) receiveOpinions(ctx context.Context, req *opinionRequest, pe
 	}
 }
 
-// GetEpochATXs fetches all ATXs published in the specified epoch from a peer.
-func (d *DataFetch) GetEpochATXs(ctx context.Context, epoch types.EpochID) error {
-	peers := d.fetcher.GetPeers()
-	if len(peers) == 0 {
-		return errNoPeers
-	}
-	peer := p2p.NoPeer
+func (d *DataFetch) pickAtxPeer(epoch types.EpochID) p2p.Peer {
+	d.mu.Lock()
+	defer d.mu.Unlock()
 	if _, ok := d.atxSynced[epoch]; !ok {
 		d.atxSynced[epoch] = map[p2p.Peer]struct{}{}
 		delete(d.atxSynced, epoch-1)
 	}
+	peers := d.fetcher.GetPeers()
 	for _, p := range peers {
 		if _, ok := d.atxSynced[epoch][p]; !ok {
-			peer = p
-			break
+			return p
 		}
 	}
+	return p2p.NoPeer
+}
+
+func (d *DataFetch) updateAtxPeer(epoch types.EpochID, peer p2p.Peer) {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	d.atxSynced[epoch][peer] = struct{}{}
+}
+
+// GetEpochATXs fetches all ATXs published in the specified epoch from a peer.
+func (d *DataFetch) GetEpochATXs(ctx context.Context, epoch types.EpochID) error {
+	peers := d.fetcher.GetPeers()
+	if len(peers) == 0 {
+		return errNoPeers
+	}
+	peer := d.pickAtxPeer(epoch)
```

Comment:
This will call `d.fetcher.GetPeers()` again.

Reply:
fixed. thanks